requirements.txt

```diff
@@ -2,3 +2,4 @@ pyyaml
 mxnet==1.1.0
 numpy>=1.12
 typing
+pillow
```

Comment:
this needs to be added to the CUDA requirements files as well.

Reply:
can we make this an optional dependency and maybe have a try block around the import with a message that the user should install pillow to use the image captioning functionality?

Reply:
[Pillow](https://pillow.readthedocs.io/en/5.1.x/handbook/overview.html) is used for image handling (in this module we use: reading, resizing, cropping). I would prefer to have it installed automatically. 
I would prefer @fhieber recommendation above: "it make sense to think about installation flags in `setup.py` to control whether a user wants to install the image captioning module or not". Any idea about how we can do this?

Reply:
Sure, I get that. However, I do quite like the fact that Sockeye by default has very few dependencies. I was thinking of something along the lines of:
```
try:
    from PIL import ...
except ImportError as e:
    logger.warn("Please install pillow to use the image_captioning module.")
    raise e
```

Reply:
I agree with you to not have too many dependencies. However, it you do not have it installed, the tests will fail. How about having your try-except block and add pillow as `tests_require` in `setup.py`? Do you think that it would work?